iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -474,6 +480,43 @@ private static PartitionSpec spec(Configuration configuration, Schema schema, Pr
     }
   }
 
+  private void handleReplaceColumns(org.apache.hadoop.hive.metastore.api.Table hmsTable) throws MetaException {
+    HiveSchemaUtil.SchemaDifference schemaDifference = HiveSchemaUtil.getSchemaDiff(hmsTable.getSd().getCols(),
+        HiveSchemaUtil.convert(icebergTable.schema()), true);
+    if (!schemaDifference.isEmpty()) {
+      updateSchema = icebergTable.updateSchema();
+    } else {
+      // we should get here if the user restated the exactly the existing columns in the REPLACE COLUMNS command
+      LOG.info("Found no difference between new and old schema for ALTER TABLE REPLACE COLUMNS for" +
+          " table: {}. There will be no Iceberg commit.", hmsTable.getTableName());
```

Comment:
nit: could place an explicit return statement here. In case any future change tweak the code below, someone could miss this path having to yield to a no-op.

Reply:
Done